ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+                    camp->BytesPerField, (const char *)valor);
+        }
+    }
+
+    memcpy(registre + camp->AcumulatedBytes, hMiraMonLayer->szStringToOperate,
+           camp->BytesPerField);
+    return 0;
+}
+
+// Gets the n-th value of the format (number_of_values:val1,val2,...,valN)
+char *MMGetNFieldValue(const char *pszStringList, GUInt32 nIRecord)
+{
+    char *p, *q;
+    GUInt32 nNValues, nIValues;
+    char *pszAux;
+
+    if (!pszStringList)
+        return nullptr;
+
+    pszAux = strdup_function(pszStringList);
+    p = strstr(pszAux, "(");
+    if (!p)
+        return nullptr;
+    p++;
+    if (!p)
+        return nullptr;
+    q = strstr(p, ":");
+    p[(ptrdiff_t)q - (ptrdiff_t)p] = '\0';
+    nNValues = atoi(p);
+    if (nIRecord > nNValues)
+        return nullptr;
+
+    q++;
+    nIValues = 0;
+    while (nIValues <= nIRecord)
+    {
+        if (!q)
+            return nullptr;
+        p = strstr(q, ",");
+        if (!p)
+        {
+            p = strstr(q, ")");
+            if (!p)
+                return nullptr;
+            q[(ptrdiff_t)p - (ptrdiff_t)q] = '\0';
+            return q;
+        }
+        if (nIValues == nIRecord)
+        {
+            p = strstr(q, ",");
+            q[(ptrdiff_t)p - (ptrdiff_t)q] = '\0';
+
+            return q;
+        }
+        q = p + 1;
+    }
+
+    return q;
+}
+
+static int MMAddFeatureRecordToMMDB(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                                    struct MiraMonFeature *hMMFeature,
+                                    struct MMAdmDatabase *pMMAdmDB,
+                                    char *pszRecordOnCourse,
+                                    struct MM_FLUSH_INFO *pFlushRecList,
+                                    MM_EXT_DBF_N_RECORDS *nNumRecords,
+                                    MM_EXT_DBF_N_FIELDS nNumPrivateMMField)
+{
+    MM_EXT_DBF_N_MULTIPLE_RECORDS nIRecord;
+    MM_EXT_DBF_N_FIELDS nIField;
+    struct MM_DATA_BASE_XP *pBD_XP = nullptr;
+
+    if (!hMiraMonLayer)
+        return 1;
+
+    if (!hMMFeature)
+        return 1;
+
+    pBD_XP = pMMAdmDB->pMMBDXP;
+    for (nIRecord = 0; nIRecord < hMMFeature->nNumMRecords; nIRecord++)
+    {
+        for (nIField = 0; nIField < hMMFeature->pRecords[nIRecord].nNumField;
+             nIField++)
+        {
+            // A field with no valid value is written as blank
+            if (!hMMFeature->pRecords[nIRecord].pField[nIField].bIsValid)
+            {
+                MM_ACUMULATED_BYTES_TYPE_DBF i = 0;
+                while (
+                    i <
+                    pBD_XP->pField[nIField + nNumPrivateMMField].BytesPerField)
+                {
+                    memcpy(pszRecordOnCourse +
+                               pBD_XP->pField[nIField + nNumPrivateMMField]
+                                   .AcumulatedBytes +
+                               i,
+                           " ", 1);
+                    i++;
+                }
```

Comment:
could likely be simplified as:
```suggestion
                memset(pszRecordOnCourse +
                               pBD_XP->pField[nIField + nNumPrivateMMField]
                                   .AcumulatedBytes, ' ', pBD_XP->pField[nIField + nNumPrivateMMField].BytesPerField);
```

Reply:
Done